ts/features/cieLogin/components/CieIdLoginWebView.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { openCieIdApp } from "@pagopa/io-react-native-cieid";
+import { Alert, Linking, Platform, StyleSheet } from "react-native";
+import WebView, { type WebViewNavigation } from "react-native-webview";
+import { SafeAreaView } from "react-native-safe-area-context";
+import _isEqual from "lodash/isEqual";
+import { useIONavigation } from "../../../navigation/params/AppParamsList";
+import { getCieIDLoginUri, SpidLevel } from "../utils";
+import { useLollipopLoginSource } from "../../lollipop/hooks/useLollipopLoginSource";
+import LoadingSpinnerOverlay from "../../../components/LoadingSpinnerOverlay";
+import { useIODispatch, useIOSelector } from "../../../store/hooks";
+import { loginSuccess } from "../../../store/actions/authentication";
+import { SessionToken } from "../../../types/SessionToken";
+import ROUTES from "../../../navigation/routes";
+import { loggedInAuthSelector } from "../../../store/reducers/authentication";
+import { IdpSuccessfulAuthentication } from "../../../components/IdpSuccessfulAuthentication";
+
+export type WebViewLoginNavigationProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const iOSUserAgent =
+  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";
+const defaultUserAgent = Platform.select({
+  ios: iOSUserAgent,
+  default: undefined
+});
+
+const originSchemasWhiteList = ["https://*", "iologin://*"];
+
+export type CieIdLoginProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
+  const navigation = useIONavigation();
+  const webView = useRef<WebView>(null);
+  const dispatch = useIODispatch();
+  const [authenticatedUrl, setAuthenticatedUrl] = useState<string | null>(null);
+  const loggedInAuth = useIOSelector(loggedInAuthSelector, _isEqual);
+  const loginUri = getCieIDLoginUri(spidLevel, isUat);
+
+  const navigateToCieIdAuthenticationError = useCallback(() => {
+    navigation.navigate(ROUTES.AUTHENTICATION, {
+      screen: ROUTES.AUTHENTICATION_CIE_ID_ERROR
+    });
+  }, [navigation]);
+
+  const { shouldBlockUrlNavigationWhileCheckingLollipop, webviewSource } =
+    useLollipopLoginSource(navigateToCieIdAuthenticationError, loginUri);
+
+  useEffect(() => {
+    // https://reactnative.dev/docs/linking#open-links-and-deep-links-universal-links
+    const urlListenerSubscription = Linking.addEventListener(
+      "url",
+      ({ url }) => {
+        // if the url is of this format: iologincie:https://idserver.servizicie.interno.gov.it/idp/login/livello2mobile?value=e1s2
+        // extract the part after iologincie: and dispatch the action to handle the login
+        if (url.startsWith("iologincie:")) {
```

Comment:
What about adding `iologincie:` to a constant? What do you think?

Reply:
I agree! Thanks!